R-packages/evalcast/tests/testthat/test-err-measures.R

```diff
@@ -74,3 +74,115 @@ test_that("Error when actual_value is the wrong length", {
   expect_error(absolute_error(quantiles, value, actual_value),
                regexp = "actual_value")
 })
+
+## These next few wis tests come from github.com/epiforecasts/scoringutils
+## They ensure that we match the covidHubUtils calculations 
+## (implemented be scoringutils::interval_score())
+test_that("wis works, median only", {
+  y <- c(1, -15, 22)
+  m <- c(1, 2, 3)
+  quantile_probs <- 0.5
+  
+  actual <- map2_dbl(y, m, weighted_interval_score, quantile = .5)
+  expected <- abs(y - m)
+  
+  
+  expect_identical(actual, expected)
+})
+
+test_that("wis works, 1 interval only", {
+  y <- c(1, -15, 22)
+  alpha <- .5
+  
+  test_data <- tibble(loc = rep(letters[1:3], each = 2), 
+                   value = c(0,2,1,2,0,3),
+                   quantile = rep(c(.25,.75), times = 3),
+                   actual = rep(c(1,-15,22), each = 2))
+  
+  ours <- test_data %>% group_by(loc) %>% 
+    summarise(wis = weighted_interval_score(quantile, value, actual)) %>%
+    pull()
+  
+  expected <- (c(2,2,3) - c(0,1,0)) * (alpha/2) + c(0, 1-(-15), 22-3)
+  
+  expect_identical(ours, expected)
+})
+
+
+test_that("wis works, 1 interval and median", {
+  test_data <- data.frame(actual =   rep(c(1, -15, 22), times = 3),
+                          quantile = rep(c(0.25, 0.5, 0.75), each = 3),
+                          value = c(c(0, 1, 0), c(1, 2, 3), c(2, 2, 3)),
+                          loc = rep(letters[1:3], times = 3))
+  
+  ours <- test_data %>% group_by(loc) %>% 
+    summarise(wis = weighted_interval_score(quantile, value, actual)) %>%
+    pull()
+  
+  # test_data <- data.frame(true_value =   rep(c(1, -15, 22), times = 3),
+  #                         quantile = rep(c(0.25, 0.5, 0.75), each = 3),
+  #                         prediction = c(c(0, 1, 0), c(1, 2, 3), c(2, 2, 3)),
+  #                         model = c("model1"),
+  #                         date = rep(1:3, times = 3))
+  # 
+  # eval <- scoringutils::eval_forecasts(
+  #   test_data,
+  #   interval_score_arguments = list(count_median_twice = FALSE))
+  # exp_wis <- eval$interval_score
```

Comment:
Is this code supposed to be used?  Would prefer not to check in commented out code

Reply:
Not used, but I took this from the other package that does these calculations for the Hub. This is to ensure that our scores are the same as theirs. I think it's good to have it in case. But I didn't want to import their package just for the tests.

Reply:
Can you pull this out into a separate test that is skipped or otherwise ignored then?

Reply:
done